src/Components/signUp/SignUp.js

```diff
@@ -0,0 +1,236 @@
+import { Fragment, useState, useEffect } from "react";
+import { Navigate } from "react-router-dom";
+
+import { Button, TextField, InputAdornment } from "@mui/material";
+import VisibilityIcon from "@mui/icons-material/Visibility";
+import { VisibilityOff } from "@mui/icons-material";
+
+import Image from "../image/ImageContainer";
+import { styles } from "../../utils";
+import "./styleSignUp.scss";
+const textFieldBorderStyle = {
+  "& .MuiInputLabel-root": {},
+  borderBottom: "1px solid rgb(179, 175, 177)",
+};
+
+const SignUp = () => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [redirectToLogin, setRedirectToLogin] = useState(false);
+  const [showPassword, setShowPassword] = useState(false);
+  const [showRepeatPassword, setShowRepeatPassword] = useState(false);
+  const [nameUser, setNameUser] = useState("");
+  const [lastNameUser, setLastNameUser] = useState("");
+  const [emailUser, setEmailUser] = useState("");
+  const [passwordUser, setPasswordUser] = useState("");
+  const [repeatPasswordUser, setRepeatPasswordUser] = useState("");
+  const [messageSignUp, setMessageSignUp] = useState({
+    text: "Enter your detalls below.",
+    messageColor: { color: "rgb(150, 157, 166)" },
+  });
+
+  const { classes } = styles();
+
+  useEffect(() => {
+    const userInformation = JSON.parse(localStorage.getItem("userInformation"));
+    if (userInformation && userInformation.isLoggedIn) {
+      setIsLoggedIn(true);
+    }
+  }, []);
+
+  const enterDashboardFunction = () => {
+    const userInformation = JSON.parse(localStorage.getItem("userInformation"));
+    if (userInformation && userInformation.email === emailUser) {
+      theUserExistsMessage();
+    } else if (passwordUser && passwordUser === repeatPasswordUser) {
+      saveUserInformation();
+      setIsLoggedIn(true);
+    } else {
+      passWordNotMatchMessage();
+    }
+  };
+
+  const theUserExistsMessage = () => {
+    setMessageSignUp({
+      text: "The user already exists. Log In.",
+      messageColor: { color: "rgb(237, 85, 151)" },
```

Comment:
we can create another variable for this as we did with `textFieldBorderStyle` and move it to a file called `materialStyles.js` see the comment above